sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/DAOModule.java

```diff
@@ -18,27 +23,122 @@
+
+        int migrationPhase = configuration.getMigrationPhase();
+        checkArgument(migrationPhase >= 0 && migrationPhase <= 3, "Invalid Migration Phase");
+
+        if (migrationPhase == 0 || migrationPhase == 1) {
+            return new CqlDataReaderDAO(delegate, placementCache, driverConfig, changeEncoder, metricRegistry);
+        }
+
+        return new CqlBlockedDataReaderDAO(delegate, placementCache, driverConfig, changeEncoder, metricRegistry, daoUtils, prefixLength);
+    }
+
+    @Provides
+    @Singleton
+    @CqlReaderDAODelegate
+    DataReaderDAO provideCqlReaderDAODelegate(DataStoreConfiguration configuration, PlacementCache placementCache,
+                                              ChangeEncoder changeEncoder, MetricRegistry metricRegistry, DAOUtils daoUtils,
+                                              @PrefixLength int prefixLength) {
+        int migrationPhase = configuration.getMigrationPhase();
+        checkArgument(migrationPhase >= 0 && migrationPhase <= 3, "Invalid Migration Phase");
+
+        if (migrationPhase == 0 || migrationPhase == 1) {
+            return new AstyanaxDataReaderDAO(placementCache, changeEncoder, metricRegistry);
+        }
+
+        return new AstyanaxBlockedDataReaderDAO(placementCache, changeEncoder, metricRegistry, daoUtils, prefixLength);
+    }
+
+    @Provides
+    @Singleton
+    DataCopyDAO provideDataCopyDAO(DataStoreConfiguration configuration, PlacementCache placementCache,
+                                   ChangeEncoder changeEncoder, MetricRegistry metricRegistry, DAOUtils daoUtils,
+                                   @PrefixLength int prefixLength) {
+
+        int migrationPhase = configuration.getMigrationPhase();
+        checkArgument(migrationPhase >= 0 && migrationPhase <= 3, "Invalid Migration Phase");
+
+        if (migrationPhase == 0 || migrationPhase == 1) {
+            return new AstyanaxDataReaderDAO(placementCache, changeEncoder, metricRegistry);
+        }
+
+        return new AstyanaxBlockedDataReaderDAO(placementCache, changeEncoder, metricRegistry, daoUtils, prefixLength);
+
+    }
+
+    @Provides
+    @Singleton
+    AstyanaxKeyScanner provideAstyanaxKeyScanner(DataStoreConfiguration configuration, PlacementCache placementCache,
+                                                       ChangeEncoder changeEncoder, MetricRegistry metricRegistry, DAOUtils daoUtils,
+                                                       @PrefixLength int prefixLength) {
+
+        int migrationPhase = configuration.getMigrationPhase();
+        checkArgument(migrationPhase >= 0 && migrationPhase <= 3, "Invalid Migration Phase");
+
+        if (migrationPhase == 0 || migrationPhase == 1) {
+            return new AstyanaxDataReaderDAO(placementCache, changeEncoder, metricRegistry);
+        }
+
+        return new AstyanaxBlockedDataReaderDAO(placementCache, changeEncoder, metricRegistry, daoUtils, prefixLength);
```

Comment:
Same comment about using Providers as from the provider method for `provideDataReaderDAO()`.

Reply:
done